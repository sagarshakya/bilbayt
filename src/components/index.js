export * from './vertical-spacer';
export * from './horizontal-spacer';
export * from './TextInput';
