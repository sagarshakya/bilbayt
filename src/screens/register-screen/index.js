import React, {useState} from 'react';
import {View, Button, Alert} from 'react-native';
import AsyncStorage from '@react-native-async-storage/async-storage';

import {axiosInstance} from '../../config/axios-config';

import {TextInput, VerticalSpacer} from '../../components';

import styles from './style';

export const RegisterScreen = () => {
  const [username, setUsername] = useState('');
  const [password, setPassword] = useState('');
  const [fullName, setFullName] = useState('');

  const onSubmit = () => {
    axiosInstance
      .post('/users', {
        username,
        password,
        extra: {
          fullName,
        },
      })
      .then(async response => {
        try {
          const {access_token: accessToken, id_token: idToken} = response.data;

          await AsyncStorage.multiSet([
            ['@app_access_token', accessToken],
            ['@app_id_token', idToken],
          ]);

          Alert.alert(
            'success',
            'You have been registered and tokens added to AsyncStorage',
          );
        } catch (e) {
          Alert.alert('Error', 'Registration failed, please try.');
        }
      })
      .catch(error => {
        Alert.alert('Error', error.response.data);
      });
  };

  return (
    <View style={styles.mainContainer}>
      <View style={styles.formContainer}>
        <TextInput
          label="Username"
          value={username}
          onChangeText={value => setUsername(value)}
          placeholder="Enter a username"
          autoCapitalize="none"
          returnKeyType="next"
        />
        <VerticalSpacer />
        <TextInput
          label="Password"
          value={password}
          onChangeText={value => setPassword(value)}
          placeholder="Enter a password"
          secureTextEntry
          returnKeyType="next"
        />
        <VerticalSpacer />
        <TextInput
          label="Full Name"
          value={fullName}
          onChangeText={value => setFullName(value)}
          placeholder="Enter your Full Name"
          autoCapitalize="words"
          returnKeyType="done"
        />
        <VerticalSpacer />
        <Button testID="register-button" title="Register" onPress={onSubmit} />
      </View>
    </View>
  );
};
