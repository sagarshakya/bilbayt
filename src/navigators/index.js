import React from 'react';
import {NavigationContainer} from '@react-navigation/native';
import {createStackNavigator} from '@react-navigation/stack';

import {RegisterScreen} from '../screens';

const AppStack = createStackNavigator();

export const MainNavigator = () => {
  return (
    <NavigationContainer>
      <AppStack.Navigator
        screenOptions={{
          headerBackTitleVisible: false,
        }}>
        <AppStack.Screen name="Register" component={RegisterScreen} />
      </AppStack.Navigator>
    </NavigationContainer>
  );
};
//
