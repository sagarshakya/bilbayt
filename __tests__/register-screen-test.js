import React from 'react';
import {render} from '@testing-library/react-native';

import {RegisterScreen} from '../src/screens';

describe('RegisterScreen component', () => {
  it('renders Username, Password and Full name labels', async () => {
    const {getByText} = render(<RegisterScreen />);

    const usernameLabel = await getByText('Username');
    const passwordLabel = await getByText('Password');
    const fullNameLabel = await getByText('Full Name');

    expect(usernameLabel).toBeTruthy();
    expect(passwordLabel).toBeTruthy();
    expect(fullNameLabel).toBeTruthy();
  });

  it('renders Username, Password and Full name input fields', async () => {
    const {getByPlaceholderText} = render(<RegisterScreen />);

    const usernameInput = await getByPlaceholderText('Enter a username');
    const passwordInput = await getByPlaceholderText('Enter a password');
    const fullNameInput = await getByPlaceholderText('Enter your Full Name');

    expect(usernameInput).toBeTruthy();
    expect(passwordInput).toBeTruthy();
    expect(fullNameInput).toBeTruthy();
  });

  it('renders Register Button', async () => {
    const {getByTestId} = render(<RegisterScreen />);

    const registerButton = await getByTestId('register-button');

    expect(registerButton).toBeTruthy();
  });
});
