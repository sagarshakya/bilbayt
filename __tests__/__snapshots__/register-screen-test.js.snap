// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegisterScreen component renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
      "padding": 15,
    }
  }
>
  <View
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <View>
      <Text>
        Username
      </Text>
      <View
        style={
          Object {
            "height": 8,
          }
        }
      />
      <TextInput
        allowFontScaling={true}
        autoCapitalize="none"
        onChangeText={[Function]}
        placeholder="Enter a username"
        rejectResponderTermination={true}
        returnKeyType="next"
        style={
          Object {
            "borderWidth": 1,
            "padding": 4,
          }
        }
        textAlignVertical="center"
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
    <View
      style={
        Object {
          "height": 8,
        }
      }
    />
    <View>
      <Text>
        Password
      </Text>
      <View
        style={
          Object {
            "height": 8,
          }
        }
      />
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        placeholder="Enter a password"
        rejectResponderTermination={true}
        returnKeyType="next"
        secureTextEntry={true}
        style={
          Object {
            "borderWidth": 1,
            "padding": 4,
          }
        }
        textAlignVertical="center"
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
    <View
      style={
        Object {
          "height": 8,
        }
      }
    />
    <View>
      <Text>
        Full Name
      </Text>
      <View
        style={
          Object {
            "height": 8,
          }
        }
      />
      <TextInput
        allowFontScaling={true}
        autoCapitalize="words"
        onChangeText={[Function]}
        placeholder="Enter your Full Name"
        rejectResponderTermination={true}
        returnKeyType="done"
        style={
          Object {
            "borderWidth": 1,
            "padding": 4,
          }
        }
        textAlignVertical="center"
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
    <View
      style={
        Object {
          "height": 8,
        }
      }
    />
    <View
      accessibilityRole="button"
      accessibilityState={Object {}}
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Register
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
      "padding": 15,
    }
  }
>
  <View
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <View>
      <Text>
        Username
      </Text>
      <View
        style={
          Object {
            "height": 8,
          }
        }
      />
      <TextInput
        allowFontScaling={true}
        autoCapitalize="none"
        onChangeText={[Function]}
        placeholder="Enter a username"
        rejectResponderTermination={true}
        returnKeyType="next"
        style={
          Object {
            "borderWidth": 1,
            "padding": 4,
          }
        }
        textAlignVertical="center"
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
    <View
      style={
        Object {
          "height": 8,
        }
      }
    />
    <View>
      <Text>
        Password
      </Text>
      <View
        style={
          Object {
            "height": 8,
          }
        }
      />
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        placeholder="Enter a password"
        rejectResponderTermination={true}
        returnKeyType="next"
        secureTextEntry={true}
        style={
          Object {
            "borderWidth": 1,
            "padding": 4,
          }
        }
        textAlignVertical="center"
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
    <View
      style={
        Object {
          "height": 8,
        }
      }
    />
    <View>
      <Text>
        Full Name
      </Text>
      <View
        style={
          Object {
            "height": 8,
          }
        }
      />
      <TextInput
        allowFontScaling={true}
        autoCapitalize="words"
        onChangeText={[Function]}
        placeholder="Enter your Full Name"
        rejectResponderTermination={true}
        returnKeyType="done"
        style={
          Object {
            "borderWidth": 1,
            "padding": 4,
          }
        }
        textAlignVertical="center"
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
    <View
      style={
        Object {
          "height": 8,
        }
      }
    />
    <View
      accessibilityRole="button"
      accessibilityState={Object {}}
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Register
        </Text>
      </View>
    </View>
  </View>
</View>
`;
